File: src/com/crossbowffs/quotelock/collection/provider/QuoteCollectionContract.java
package com.crossbowffs.quotelock.collection.provider;

import android.content.ContentResolver;
import android.net.Uri;
import android.provider.BaseColumns;

import com.crossbowffs.quotelock.BuildConfig;

/**
 * @author Yubyf
 */
public class QuoteCollectionContract {
    public static final String AUTHORITY = BuildConfig.APPLICATION_ID + ".collection.provider";
    public static final Uri CONTENT_URI = Uri.parse("content://" + AUTHORITY);

    public static class Collection implements BaseColumns {
        public static final String TABLE = "collections";
        public static final Uri CONTENT_URI = Uri.withAppendedPath(QuoteCollectionContract.CONTENT_URI, TABLE);
        public static final String CONTENT_TYPE = ContentResolver.CURSOR_DIR_BASE_TYPE + "/vnd.crossbowffs.collection";
        public static final String CONTENT_ITEM_TYPE = ContentResolver.CURSOR_ITEM_BASE_TYPE + "/vnd.crossbowffs.collection";

        public static final String MD5 = "md5";
        public static final String TEXT = "text";
        public static final String SOURCE = "source";
        public static final String[] ALL = {
                _ID,
                MD5,
                TEXT,
                SOURCE
        };
    }
}

File: app/src/main/java/com/crossbowffs/quotelock/collections/provider/QuoteCollectionContract.kt
package com.crossbowffs.quotelock.collections.provider

import android.content.ContentResolver
import android.net.Uri
import android.provider.BaseColumns
import com.crossbowffs.quotelock.BuildConfig

/**
 * @author Yubyf
 */
object QuoteCollectionContract {
    const val AUTHORITY = BuildConfig.APPLICATION_ID + ".collection.provider"
    val CONTENT_URI: Uri = Uri.parse("content://$AUTHORITY")

    object Collections : BaseColumns {
        const val TABLE = "collections"

        @JvmField
        val CONTENT_URI: Uri = Uri.withAppendedPath(QuoteCollectionContract.CONTENT_URI, TABLE)
        const val CONTENT_TYPE =
            "${ContentResolver.CURSOR_DIR_BASE_TYPE}/vnd.crossbowffs.collection"
        const val CONTENT_ITEM_TYPE =
            "${ContentResolver.CURSOR_DIR_BASE_TYPE}/vnd.crossbowffs.collection"
        const val MD5 = "md5"
        const val TEXT = "text"
        const val SOURCE = "source"
        const val ID = BaseColumns._ID
        val ALL = arrayOf(
            ID,
            MD5,
            TEXT,
            SOURCE
        )
    }
}